packages/lockfile-lint-api/src/ParseLockfile.js

```diff
@@ -20,14 +20,14 @@ const {
  * Checks if a sample object is a valid dependency structure
  * @return boolean
  */
-function checkSampleContent (lockfile) {
-  const [sampleKey, sampleValue] = Object.entries(lockfile)[0]
+function checkSampleContent (lockfile, isYarnBerry) {
```

Comment:
new implementation may diverge further (we might discover or they might add new differences) - it'd be best if these were separate functions IMHO.  
Does anybody feel it's better to DRY it?

Reply:
@naugtur you mean to have a separate function to check for yarn's version (1, 2, 3, etc) ? I guess in this part specifically of parsing the lockfile we really just need to ensure it is a parsable one, so perhaps we simply take in the detected lockfile type and then call one of the package manager's own lib & function to parse, and as long as it doesn't throw, we're ok here.

I don't mind doing that in another new PR though if @brad-decker is up for it.

Reply:
@naugtur you mean to have a specific function to check for yarn's version (1, 2, 3, etc)